crates/astria-composer/src/searcher/mod.rs

```diff
@@ -216,111 +156,96 @@ impl Searcher {
     }
 
     /// Serializes and signs a sequencer tx from a rollup tx.
-    fn handle_pending_tx(&mut self, tx: collector::Transaction) {
+    fn bundle_pending_tx(&mut self, tx: collector::Transaction) {
         let collector::Transaction {
             chain_id,
             inner: rollup_tx,
         } = tx;
 
-        let sequencer_key = self.sequencer_key.clone();
-        let nonce = self.sequencer_nonce.clone();
-
+        // rollup transaction data serialization is a heavy compute task, so it is spawned
+        // on tokio's blocking threadpool
         self.conversion_tasks.spawn_blocking(move || {
```

Comment:
I believe this is no longer necessary

Reply:
i disagree - since collectors are ingesting transactions concurrrently there is no reason serialization shouldnt happen concurrently. this will eventually need to be fixed, see issue #409 for an in-depth explanation.